File: src/main/java/fi/quanfoxes/phases/FilePhase.java
package fi.quanfoxes.phases;

import java.nio.file.Files;
import java.nio.file.Paths;

import fi.quanfoxes.Bundle;
import fi.quanfoxes.Phase;
import fi.quanfoxes.Status;

public class FilePhase extends Phase {

    @Override
    public Status execute(Bundle bundle) {
        final String[] files = bundle.get("input_files", new String[] {});

        if (files.length == 0) {
            return Status.error("Please enter input files");
        }

        final String[] contents = new String[files.length];

        for (int i = 0; i < files.length; i++) {
            final int index = i;

            async(() -> {
                String file = files[index];

                try {
                    String content = Files.readString(Paths.get(file));
                    contents[index] = content;
                }
                catch (Exception e) {
                    return Status.error("Couldn't load file '%s'", file);
                }          

                return Status.OK;
            });
        }

        bundle.put("input_file_contents", contents);

        return Status.OK;
    }
}

File: Zigzag/Phases/FilePhase.cs
using System.IO;

public class FilePhase : Phase
{
	public override Status Execute(Bundle bundle)
	{
		string[] files = bundle.Get("input_files", new string[] { });

		if (files.Length == 0)
		{
			return Status.Error("Please enter input files");
		}

		string[] contents = new string[files.Length];

		for (int i = 0; i < files.Length; i++)
		{
			int index = i;

			Async(() =>
			{
				string file = files[index];

				try
				{
					string content = File.ReadAllText(file);
					contents[index] = content.Replace("\r\n", "\n");
				}
				catch
				{
					return Status.Error("Couldn't load file '{0}'", file);
				}

				return Status.OK;
			});
		}

		bundle.Put("input_file_contents", contents);

		return Status.OK;
	}
}